src/tests/tables.test.js

```diff
@@ -3,20 +3,94 @@ const glob = require("glob");
 
 describe("duplicate indices", () => {
   it("should contain unique indices", () => {
+    let errors = [];
+    const fileIndices = {};
+
+    forEachFile((filename, entry) => {
+      if (filename in fileIndices === false) {
+        fileIndices[filename] = new Set();
+      }
+
+      if (fileIndices[filename].has(entry.index)) {
+        errors.push(`${filename}: Index '${entry.index}' already exists.`);
+      }
+
+      fileIndices[filename].add(entry.index);
+    });
+
+    expect(errors).toEqual([]);
+  });
+});
+
+describe("api references", () => {
+  it("should not contain broken links", () => {
     let errors = [];
     let files = glob.sync("src/*.json");
 
-    for (const file of files) {
-      const fileText = fs.readFileSync(file, "utf8");
-      const fileJSON = JSON.parse(fileText);
-      const indices = new Set();
-      fileJSON.forEach((entry) => {
-        if (indices.has(entry.index)) {
-          errors.push(`${file}: Index '${entry.index}' already exists.`);
+    let resources = {};
+
+    forEachFile((filename, entry) => {
+      if (entry.url !== undefined) {
+        if (entry.index === undefined) {
+          errors.push(`${filename}: Entry with URL '${entry.url}' should have an index.`);
         }
-        indices.add(entry.index);
-      });
-    }
+
+        resources[entry.url] = { index: entry.index, name: entry.name };
+      }
+    });
+
+    forEachFileRecursive((filename, entry) => {
+      if (entry.hasOwnProperty("url")) {
+        if (resources[entry.url] === undefined) {
+          errors.push(`${filename}: URL '${entry.url}' not found.`);
+        } else {
+          if (resources[entry.url].name !== undefined && resources[entry.url].name !== entry.name) {
+            errors.push(`${filename}: Name mismatch for reference to '${entry.url}', '${entry.name}' should be '${resources[entry.url].name}'`);
+          }
+
+          if (entry.index !== undefined && resources[entry.url].index !== entry.index) {
+            errors.push(`${filename}: Index mismatch for reference to '${entry.url}', '${entry.index}' should be '${resources[entry.url].index}'`);
+          }
+        }
+      }
```

Comment:
Can we also shift this to a guarded clause?

```suggestion
      if (!entry.hasOwnProperty("url")) return;
      
      if (resources[entry.url] === undefined) {
        errors.push(`${filename}: URL '${entry.url}' not found.`);
      } else {
        if (resources[entry.url].name !== undefined && resources[entry.url].name !== entry.name) {
          errors.push(`${filename}: Name mismatch for reference to '${entry.url}', '${entry.name}' should be '${resources[entry.url].name}'`);
        }

        if (entry.index !== undefined && resources[entry.url].index !== entry.index) {
          errors.push(`${filename}: Index mismatch for reference to '${entry.url}', '${entry.index}' should be '${resources[entry.url].index}'`);
        }
      }
```